politeiad/backendv2/backendv2.go

```diff
@@ -0,0 +1,350 @@
+// content was timestamped onto the decred blockchain.
+//
+// All digests are hex encoded SHA256 digests. The merkle root can be found in
+// the OP_RETURN of the specified DCR transaction.
+//
+// TxID, MerkleRoot, and Proofs will only be populated once the merkle root has
+// been included in a DCR tx and the tx has 6 confirmations. The Data field
+// will not be populated if the data has been censored.
+type Timestamp struct {
+	Data       string // JSON encoded
+	Digest     string
+	TxID       string
+	MerkleRoot string
+	Proofs     []Proof
+}
+
+// RecordTimestamps contains a Timestamp for all record data.
+type RecordTimestamps struct {
+	RecordMetadata Timestamp
+
+	Metadata map[string]map[uint32]Timestamp // [pluginID][streamID]Timestamp
+	Files    map[string]Timestamp            // map[filename]Timestamp
+}
+
+// Inventory contains the tokens of records in the inventory categorized by
+// record state and record status. Tokens are sorted by the timestamp of the
+// status change from newest to oldest.
+type Inventory struct {
+	Unvetted map[StatusT][]string
+	Vetted   map[StatusT][]string
+}
+
+// PluginSetting represents a configurable plugin setting.
+type PluginSetting struct {
+	Key   string // Name of setting
+	Value string // Value of setting
+}
+
+// Plugin describes a plugin and its settings.
+type Plugin struct {
+	ID       string
+	Settings []PluginSetting
+
+	// Identity contains the full identity that the plugin uses to
+	// create receipts, i.e. signatures of user provided data that
+	// prove the backend received and processed a plugin command.
+	Identity *identity.FullIdentity
+}
+
+// PluginError represents an error that occured during plugin execution that
+// was caused by the user.
+type PluginError struct {
+	PluginID     string
+	ErrorCode    int
+	ErrorContext string
+}
+
+// Error satisfies the error interface.
+func (e PluginError) Error() string {
+	return fmt.Sprintf("plugin '%v' error code %v",
+		e.PluginID, e.ErrorCode)
+}
+
+// Backend provides an API for interacting with records in the backend.
+type Backend interface {
+	// RecordNew creates a new record.
+	RecordNew([]MetadataStream, []File) (*Record, error)
+
+	// RecordEdit edits an existing record.
+	RecordEdit(token []byte, mdAppend, mdOverwrite []MetadataStream,
+		filesAdd []File, filesDel []string) (*Record, error)
+
+	// RecordEditMetadata edits the metadata of a record without
+	// editing any record files.
+	RecordEditMetadata(token []byte, mdAppend,
+		mdOverwrite []MetadataStream) (*Record, error)
+
+	// RecordSetStatus sets the status of a record.
+	RecordSetStatus(token []byte, s StatusT, mdAppend,
+		mdOverwrite []MetadataStream) (*Record, error)
+
+	// RecordExists returns whether a record exists.
+	RecordExists(token []byte) bool
+
+	// RecordGet retrieves a record. If no version is provided then the
+	// most recent version will be returned.
+	RecordGet(token []byte, version uint32) (*Record, error)
```

Comment:
We should get rid of single gets and always use the batched one.

Reply:
Removed on both the politeiad v2 api and backend.

https://github.com/lukebp/politeia/blob/tlog/politeiad/api/v2/v2.go#L13
https://github.com/lukebp/politeia/blob/tlog/politeiad/backendv2/backendv2.go#L278